examples/mnist/model.go

```diff
@@ -0,0 +1,94 @@
+package mnist
+
+// This file implements the baseline CNN model, including the FNN layers on top.
+
+import (
+	"github.com/gomlx/exceptions"
+	. "github.com/gomlx/gomlx/graph"
+	"github.com/gomlx/gomlx/ml/context"
+	"github.com/gomlx/gomlx/ml/layers"
+	"github.com/gomlx/gomlx/ml/layers/activations"
+	"github.com/gomlx/gomlx/ml/layers/batchnorm"
+	"github.com/gomlx/gomlx/ml/layers/fnn"
+)
+
+// CnnModelGraph builds the CNN model for our demo.
+// It returns the logit, not the predictions, which works with most losses.
+// inputs: only one tensor, with shape `[batch_size, width, height, depth]`.
+func SoftMaxModelGraph(ctx *context.Context, spec any, inputs []*Node) []*Node {
+	ctx = ctx.In("model") // Create the model by default under the "/model" scope.
+	batchSize := inputs[0].Shape().Dimensions[0]
+	embeddings := Reshape(inputs[0], batchSize, -1)
+	logit := Softmax(layers.DenseWithBias(ctx, embeddings, 1))
+	return []*Node{Squeeze(logit)}
+}
+
+// CnnModelGraph builds the CNN model for our demo.
+// It returns the logit, not the predictions, which works with most losses.
+// inputs: only one tensor, with shape `[batch_size, width, height, depth]`.
+func CnnModelGraph(ctx *context.Context, spec any, inputs []*Node) []*Node {
+	ctx = ctx.In("model") // Create the model by default under the "/model" scope.
+	embeddings := CnnEmbeddings(ctx, inputs[0])
+	logit := fnn.New(ctx.In("readout"), embeddings, 1).NumHiddenLayers(0, 0).Done()
```

Comment:
This is a FNN (Feedforward Neural Network) model, not a CnnModel. Update names ?

Also, as previous comment, for MNIST I think you want the output dimension to be 10. So the output shape shoud be `[batchSize, 10]`

Reply:
complete wrong, I just copy from dog and cat without knowing what is it.